rasterio/rio/info.py

```diff
@@ -61,12 +62,10 @@ def info(ctx, input, aspect, indent, namespace, meta_member, verbose, bidx,
     Optionally print a single metadata item as a string.
     """
     verbosity = ctx.obj.get('verbosity')
-    aws_session = ctx.obj.get('aws_session')
     mode = 'r' if (verbose or meta_member == 'stats') else 'r-'
     try:
-        with rasterio.drivers(
-                CPL_DEBUG=(verbosity > 2)), aws_session:
-            with rasterio.open(input, mode) as src:
+        with Session(CPL_DEBUG=(verbosity > 2)) as session:
+            with session.open(input, mode) as src:
```

Comment:
Here's the new proposed usage. Simpler!

Using the session's `open()` means that we can update the session with AWS credentials if `open()` finds that the provided path is an S3 URI.